CRM/Autologcleanup/Job/LogCleanup.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+class CRM_Autologcleanup_Job_LogCleanup {
+
+  /**
+   * @var string
+   */
+  private $maxAge;
+
+  /**
+   * Constructs a new LogCleanup object.
+   *
+   * @param string $maxAge
+   *  Maximum log age.
+   */
+  public function __construct(string $maxAge) {
+    $this->maxAge = $maxAge;
+  }
+
+  /**
+   * Starts the scheduled job for clearing up
+   * stale log records.
+   *
+   * @return True
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function run() {
+    $this->pruneLogTables();
+
+    return TRUE;
+  }
+
+  /**
+   * Tranforms the log age to date ago.
+   *
+   * @param string $age
+   *  The age of the log to be cleared.
+   *
+   * @throws \CRM_Core_Exception;
+   */
+  private function transformLogAge(string $age) {
+    $timestamp = strtotime("today - $age");
+    if (!$timestamp) {
+      throw new \CRM_Core_Exception("${age} is an invalid log max age");
+    }
+
+    return date('Y-m-d', $timestamp);
+  }
+
+  /**
+   * Prunes stale records from log tables.
+   *
+   * Records that have stayed loger than the max age
+   * are considered stale.
+   */
+  private function pruneLogTables() {
+    $ageAgo = $this->transformLogAge($this->maxAge);
+    $logTables = $this->getLogTables();
+
+    foreach ($logTables as $logTable) {
+      $query = CRM_Utils_SQL_Delete::from($logTable)
+        ->where('log_date < @date', ['date' => $ageAgo]);
+      $query->execute();
+    }
+  }
+
+  /**
+   * Gets all the log tables
+   *
+   * @return array
+   *   Returns an array of log tables
+   */
+  private function getLogTables() {
+    $logTables = [];
+    $loggingSchema = new \CRM_Logging_Schema();
+    $nonStandardTableNameString = $loggingSchema->getNonStandardTableNameFilterString();
```

Comment:
it would be more cleaner if you move the part that gets the dsn into its own private method.